GithubApp/Controllers/SearchViewController.swift

```diff
@@ -48,25 +72,86 @@ extension SearchViewController: UITableViewDataSource{
+        let cell: SearchTableViewCell = tableView.dequeueCell(for: indexPath)
         let personData = searchedData[indexPath.row]
         cell.backgroungImage.image = UIImage(named: "R")
-        cell.personImage.loadFrom(URLAddress: personData.owner!.avatarUrl)
         cell.nameLabel.text = personData.name
         cell.descriptionLabel.text = personData.description
+        cell.languageLabel.text = personData.languageUrl.rawValue
+        guard let imageURL = personData.owner?.avatarUrl else {
+            cell.personImage.image = UIImage(named: "Image1")
+            return cell
+        }
+        cell.personImage.loadFrom(URLAddress: imageURL)
         return cell
     }
-    
 }
 
 extension SearchViewController: UISearchBarDelegate{
     func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
         searchedData.removeAll()
         for person in data {
-            if (person.name!.hasPrefix(searchText.lowercased())){
+            guard let personName = person.name else{
+                return
+            }
+            if (personName.hasPrefix(searchText.lowercased())){
                 searchedData.append(person)
             }
         }
         self.searchView.tableView.reloadData()
     }
 }
+
+extension SearchViewController: FilterDelegate{
+    
+    // MARK: Functions 
+    func getSelectedFilters(filters: [Filter]) {
+        self.filterData.removeAll()
+        self.filterData = filters
+        self.searchedData.removeAll()
+        var isFilterApplied :Bool = true
+        for filter in filters {
+            if(filter.isSelected){
+                isFilterApplied = false
+                break
+            }
+        }
+        
+        if isFilterApplied {
+            reloadData()
+        }else{
+            for person in data {
+                switch (person.languageUrl.rawValue){
+                case languages.Python.rawValue:
+                    appendSearchedData(index: 0, person: person, filters: filters)
+                case languages.InteriorPainting.rawValue:
+                    appendSearchedData(index: 1, person: person, filters: filters)
+                case languages.Electrician.rawValue:
+                    appendSearchedData(index: 2, person: person, filters: filters)
+                case languages.InappropriateClientBehavior.rawValue:
+                    appendSearchedData(index: 3, person: person, filters: filters)
+                case languages.Gardener.rawValue:
+                    appendSearchedData(index: 4, person: person, filters: filters)
+                case languages.Carpenter.rawValue:
+                    appendSearchedData(index: 5, person: person, filters: filters)
+                case languages.HouseCleaning.rawValue:
+                    appendSearchedData(index: 6, person: person, filters: filters)
+                default:
+                    print("No such Item")
+                }
+            }
```

Comment:
you should be applying switch on enum directly. No need to use switch enum's rawValue when you have the enum directly.

Reply:
Done